set.go

```diff
@@ -194,3 +194,27 @@ func NewThreadUnsafeSet[T comparable](vals ...T) Set[T] {
 	}
 	return &s
 }
+
+// Creates and returns a new set with the given keys of the map.
+// Operations on the resulting set are thread-safe.
+func NewSetFromMapKeys[T comparable, V any](val map[T]V) Set[T] {
+	s := NewSet[T]()
+
+	for k := range val {
+		s.Add(k)
+	}
+
+	return s
+}
+
+// Creates and returns a new set with the given keys of the map.
```

Comment:
Per `godoc` convertions, these docstrings should start with the name of the func being documented.

Related:
* https://github.com/deckarep/golang-set/pull/106

Reply:
@jeffwidman - your keen eye strikes again. I need to add a linter at some point to catch these kinds of things before merge. I'll get around to it at some point.

Thanks!